lib/mix/lib/mix/compilers/erlang.ex

```diff
@@ -151,6 +149,17 @@ defmodule Mix.Compilers.Erlang do
       # Return status and diagnostics
       warnings = manifest_warnings(entries) ++ to_diagnostics(warnings, :warning)
 
+      lazy_modules_diff = fn ->
+        {changed, added} =
+          stale
+          |> Enum.map(&elem(&1, 1))
+          |> Enum.split_with(fn dest -> List.keymember?(old_entries, dest, 0) end)
+
+        modules_diff(added, changed, removed, timestamp)
+      end
+
+      Mix.Task.Compiler.notify_modules_compiled(lazy_modules_diff)
```

Comment:
We need to check that the produced artifacts by the compiler is actually .beam files. I believe it is also used to produce non .beam, no?

Reply:
Good call!

Reply:
`compile.yecc` would produce `.erl` files, which we want to ignore, and only then `compile.erlang` will produce `.beam` for these, which we announce correctly.